orchestra/todos/api.py

```diff
@@ -17,17 +21,50 @@ def add_todolist_template(todolist_template_slug, task_id):
         template=todolist_template
     )
     root_todo.save()
+
+    cond_props = {}
+    path = todolist_template.conditional_property_function.get(
+        'path', None)
+    if path:
+        try:
+            get_cond_props = locate(path)
+            cond_props = get_cond_props(task.project)
+        except Exception:
+            logger.exception('Invalid condition function path.')
     for template_todo in template_todos:
-        _add_template_todo(template_todo, todolist_template, root_todo, task)
+        _add_template_todo(
+            template_todo, todolist_template, root_todo, task, cond_props)
 
 
-def _add_template_todo(template_todo, todolist_template, parent_todo, task):
-    todo = Todo(
-        task=task,
-        description=template_todo['description'],
-        template=todolist_template,
-        parent_todo=parent_todo
-    )
-    todo.save()
-    for template_todo_item in template_todo.get('items', []):
-        _add_template_todo(template_todo_item, todolist_template, todo, task)
+def _to_exclude(props, conditions=[]):
```

Comment:
https://stackoverflow.com/questions/5712904/empty-dictionary-as-default-value-for-keyword-argument-in-python-function-dicti :)

you can do this:
```
def _to_exclude(props, conditions=None):
    for condition in conditions or []:

Reply:
alternatively, you already pass in `[]` below, so just make `conditions` required ;)